deployment/ccip/changeset/cs_deploy_token_pools.go

```diff
@@ -0,0 +1,497 @@
+		}
+	} else {
+		if rateLimiterConfig.Rate.Cmp(zero()) != 0 || rateLimiterConfig.Capacity.Cmp(zero()) != 0 {
+			return errors.New("rate and capacity must be 0")
+		}
+	}
+	return nil
+}
+
+// RemoteChainsConfig defines rate limits for remote chains.
+type RemoteChains map[uint64]RateLimiterConfig
+
+func (rc RemoteChains) Validate() error {
+	for chainSelector, chainConfig := range rc {
+		if err := validateRateLimiterConfig(chainConfig.Inbound); err != nil {
+			return fmt.Errorf("validation of inbound rate limiter config for remote chain with selector %d failed: %w", chainSelector, err)
+		}
+		if err := validateRateLimiterConfig(chainConfig.Outbound); err != nil {
+			return fmt.Errorf("validation of outbound rate limiter config for remote chain with selector %d failed: %w", chainSelector, err)
+		}
+	}
+	return nil
+}
+
+// BaseTokenPoolInput defines all the information required of the user to configure new and existing pools.
+type BaseTokenPoolInput struct {
+	RemoteChainsToAdd RemoteChains
+	TokenAddress      common.Address
+	ExternalAdmin     common.Address
+}
+
+func (t BaseTokenPoolInput) Validate() error {
+	if err := t.RemoteChainsToAdd.Validate(); err != nil {
+		return fmt.Errorf("failed to validate remote chains config: %w", err)
+	}
+	return nil
+}
+
+// NewTokenPoolInput defines all information required of the user to deploy and configure a new token pool.
+type NewTokenPoolInput struct {
+	BaseTokenPoolInput
+	Type               deployment.ContractType
+	LocalTokenDecimals uint8
+	AllowList          []common.Address
+	AcceptLiquidity    *bool
+}
+
+func (t NewTokenPoolInput) Validate() error {
+	if err := t.RemoteChainsToAdd.Validate(); err != nil {
+		return fmt.Errorf("failed to validate remote chains config: %w", err)
+	}
+	if t.Type != BurnMintTokenPool && t.Type != BurnFromMintTokenPool && t.Type != BurnWithFromMintTokenPool && t.Type != LockReleaseTokenPool {
+		return fmt.Errorf("%s is not a valid token pool type", t.Type)
+	}
+	if t.Type == LockReleaseTokenPool && t.AcceptLiquidity == nil {
+		return errors.New("accept liquidity must be defined for lock release pools")
+	}
+	if t.Type != LockReleaseTokenPool && t.AcceptLiquidity != nil {
+		return errors.New("accept liquidity must be nil for burn mint pools")
+	}
+	return nil
+}
+
+// DeployTokenPoolContractsConfig is the configuration for the DeployTokenPoolContracts changeset.
+type DeployTokenPoolContractsConfig struct {
+	Symbol              TokenSymbol
+	TimelockDelay       time.Duration
+	ExistingPoolUpdates map[uint64]BaseTokenPoolInput
+	NewPools            map[uint64]NewTokenPoolInput
+}
+
+func (c DeployTokenPoolContractsConfig) Validate() error {
+	seenChains := make(map[uint64]struct{})
+	for chainSelector, chainConfig := range c.ExistingPoolUpdates {
+		seenChains[chainSelector] = struct{}{}
+		if err := chainConfig.Validate(); err != nil {
+			return fmt.Errorf("chain with selector %d is invalid: %w", chainSelector, err)
+		}
+	}
```

Comment:
Need to include [IsValidChainSelector](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/helpers.go#L182) check